components/basic-filters/src/main/java/org/datacleaner/beans/filter/StringContainsFilter.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Free Software Foundation, Inc.
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.beans.filter;
+
+import javax.inject.Named;
+
+import org.datacleaner.api.Categorized;
+import org.datacleaner.api.Configured;
+import org.datacleaner.api.Description;
+import org.datacleaner.api.Filter;
+import org.datacleaner.api.InputColumn;
+import org.datacleaner.api.InputRow;
+import org.datacleaner.components.categories.FilterCategory;
+
+@Named("String contains filter")
+@Description("Check if your string values contain specific substrings in order to filter them in or out.")
+@Categorized(FilterCategory.class)
+public class StringContainsFilter implements Filter<StringContainsFilter.Category> {
+
+    public enum Category {
+        @Description("Outcome when a substring is matched/contained in the input string.")
+        MATCHED,
+
+        @Description("Outcome when none of the substrings are matched/contained in the input string.")
+        UNMATCHED
+    }
+
+    @Configured(order = 1)
+    InputColumn<String> column;
+
+    @Configured(order = 2)
+    @Description("Substring(s) to look for")
+    String[] substrings;
+
+    @Configured(order = 3)
+    boolean caseSensitive = false;
+
+    @Override
+    public Category categorize(final InputRow inputRow) {
+        final String value = inputRow.getValue(column);
+        return categorize(value);
+    }
+
+    protected Category categorize(String value) {
```

Comment:
Why is this method protected and not just private? Do you expect this class to be extended and the method to be overridden?

Also I would like the `value` parameter to be final (which implies that some of the internal logic of the method has to be changed.)

Reply:
I made it protected actually because I was thinking of testing it without having to make MockInputRows and such. But I'm gonna change it to private and test it via the public interface.